src/common/datatypes/HostAddr.h

```diff
@@ -40,11 +42,21 @@ struct HostAddr {
     return os.str();
   }
 
+  HostAddr& operator=(const HostAddr& rhs);
+
   bool operator==(const HostAddr& rhs) const;
 
   bool operator!=(const HostAddr& rhs) const;
 
   bool operator<(const HostAddr& rhs) const;
+
+  static HostAddr fromString(const std::string& str) {
+    HostAddr ha;
+    auto pos = str.find(":");
+    ha.host = str.substr(1, pos - 2);
```

Comment:
add some Check here ? this method may core if pass an invalid str.

Reply:
done